### DISCUSSION TITLE
Open telemetry logs do not include exceptions or structured logs

### DISCUSSION THREAD

**Discussion by vince-nyanga** [datalust/seq-tickets#2119] [Help]
I have configured open telemetry logging to write to Seq like this:
```cs
 logging.AddOpenTelemetry(openTelemetryLoggerOptions =>
            {

                openTelemetryLoggerOptions.SetResourceBuilder(
                    ResourceBuilder.CreateEmpty()
                        .AddService(serviceName: Telemetry.ActivityName));

                openTelemetryLoggerOptions.IncludeScopes = true;

                openTelemetryLoggerOptions.AddOtlpExporter(opt =>
                {
                    opt.Endpoint = new Uri($"{seqServerUrl}/ingest/otlp/v1/logs");
                    opt.Protocol = OtlpExportProtocol.HttpProtobuf;
                });
            });
```

I can see the logs coming through to Seq. However, where I try to log  an error that includes an exception, I'm not seeing the exception details in Seq:

```cs
_logger.LogError(ex, "Something happened");
```

And when I have a complex object in the log, it just renders the `ToString()` variant of the log:

```cs
_logger.LogInformation("This is a complex object: {@MyObject}", myObject);
```

I'm not sure what I could might be missing.

**Comment by KodrAus** [comment]
Thanks for the report @vince-nyanga. We're taking a look at this and will let you know what we find.

**Comment by nblumhardt** [comment]
The .NET OTLP exporter from the OpenTelemetry project doesn't currently include exception details in log events (this [is merged](https://github.com/open-telemetry/opentelemetry-dotnet/pull/5258), but still [unreleased](https://github.com/open-telemetry/opentelemetry-dotnet/blob/main/src/OpenTelemetry.Exporter.OpenTelemetryProtocol/CHANGELOG.md#unreleased)).

You can enable the experimental option to include these by setting an environment variable:

```
OTEL_DOTNET_EXPERIMENTAL_OTLP_EMIT_EXCEPTION_LOG_ATTRIBUTES=True
```

Regarding the `@` operator, I don't believe this is supported by the MEL/OpenTelemetry back-end; you'd need to use either Serilog or NLog to get this functionality.

Hope this helps, and if you need more info or I've overlooked anything, please just let me know.
